src/cmd/tools/read_data_files/main/main.go

```diff
@@ -248,20 +254,63 @@ func main() {
 	}
 }
 
+func getVolumeNumber(pathPrefix string, namespace string, shard int, blockStart string) (int, error) {
+	nsID := ident.StringID(namespace)
+	path := fs.NamespaceDataDirPath(pathPrefix, nsID)
+	files, err := ioutil.ReadDir(path + "/" + strconv.Itoa(shard))
+	if err != nil {
+		return 0, fmt.Errorf("failed reading data directory: %w", err)
+	}
+
+	volumeSet := make(map[int]struct{})
+	maxVolumeNum := 0
+	for _, f := range files {
+		if f.IsDir() {
+			continue
+		}
+		splittedFileName := strings.Split(f.Name(), "-")
+		if len(splittedFileName) != fs.MaxDelimNum {
+			fmt.Errorf("failed parsing volume number from file name: %s, not enough segments", f.Name())
+		} else {
+			if splittedFileName[1] == blockStart {
+				volumeNum, err := strconv.Atoi(splittedFileName[2])
+				if err != nil {
+					fmt.Errorf("failed parsing volume directory: %w", err)
+				} else {
+					volumeSet[volumeNum] = struct{}{}
+					if volumeNum > maxVolumeNum {
+						maxVolumeNum = volumeNum
+					}
+				}
+			} else {
+				continue
+			}
+		}
+	}
+
+	volumeNums := make([]int, 0, len(volumeSet))
+	for volume := range volumeSet {
+		volumeNums = append(volumeNums, volume)
+	}
```

Comment:
Should delete it. It doesn't have any effect.

Reply:
fixed